piper/leddialog.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (C) 2017 Jente Hidskes <hjdskes@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+from .gi_composites import GtkTemplate
+from .ratbagd import RatbagdLed
+
+import gi
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gdk, Gtk
+
+
+@GtkTemplate(ui="/org/freedesktop/Piper/ui/LedDialog.ui")
+class LedDialog(Gtk.Dialog):
+    """A Gtk.Dialog subclass to implement the dialog that shows the
+    configuration options for the LED effects."""
+
+    __gtype_name__ = "LedDialog"
+
+    stack = GtkTemplate.Child()
+    colorchooser = GtkTemplate.Child()
+    colorbutton = GtkTemplate.Child()
+    adjustment_brightness = GtkTemplate.Child()
+    adjustment_effect_rate = GtkTemplate.Child()
+
+    def __init__(self, ratbagd_led, *args, **kwargs):
+        """Instantiates a new LedDialog.
+
+        @param ratbagd_led The LED to configure, as ratbagd.RatbagdLed.
+        """
+        Gtk.Dialog.__init__(self, *args, **kwargs)
+        self.init_template()
+        self._led = ratbagd_led
+
+        mode = self._led.mode
+        if mode == RatbagdLed.MODE_ON:
+            self.stack.set_visible_child_name("solid")
+        elif mode == RatbagdLed.MODE_CYCLE:
+            self.stack.set_visible_child_name("cycle")
+        elif mode == RatbagdLed.MODE_BREATHING:
+            self.stack.set_visible_child_name("breathing")
+        elif mode == RatbagdLed.MODE_OFF:
+            self.stack.set_visible_child_name("off")
```

Comment:
I think you could/should use a dictionary [string] = enum here for the mode mappings. In `_on_stack_page_changed` you can do a simple lookup, here you either invert the dict or simply loop through `.keys()` until you find the value.